src/lib/djerba/plugins/pwgs/analysis/plugin.py

```diff
@@ -54,28 +56,36 @@ def extract(self, config):
         mrdetect_results = pwgs_tools.preprocess_results(self, config[self.identifier][pc.RESULTS_FILE])
         hbc_results = self.preprocess_hbc(config[self.identifier][pc.HBC_FILE])
         reads_detected = self.preprocess_vaf(config[self.identifier][pc.VAF_FILE])
-        pwgs_base64 = self.write_pwgs_plot(config[self.identifier][pc.HBC_FILE], 
+        pwgs_base64 = self.write_pwgs_plot(config[self.identifier][pc.HBC_FILE],
                                            config[self.identifier][pc.VAF_FILE],
-                                           output_dir = self.workspace.print_location())
+                                           output_dir=self.workspace.print_location())
         self.logger.info("PWGS ANALYSIS: Finished preprocessing files")
-        data = self.get_starting_plugin_data(wrapper, self.PLUGIN_VERSION)       
-        results =  {
-                pc.CTDNA_OUTCOME: mrdetect_results[pc.CTDNA_OUTCOME],
-                pc.SIGNIFICANCE: mrdetect_results[pc.SIGNIFICANCE],
-                pc.TUMOUR_FRACTION_READS: float('%.1E' % Decimal( reads_detected*100 / hbc_results[pc.READS_CHECKED] )),
-                pc.SITES_CHECKED: hbc_results[pc.SITES_CHECKED],
-                pc.READS_CHECKED: hbc_results[pc.READS_CHECKED],
-                pc.SITES_DETECTED: hbc_results[pc.SITES_DETECTED],
-                pc.READS_DETECTED: reads_detected,
-                pc.PVALUE: mrdetect_results[pc.PVALUE],
-                pc.COHORT_N: hbc_results[pc.COHORT_N],
-                'pwgs_base64': pwgs_base64,
-                'files': {
-                    'results_file': config[self.identifier][pc.RESULTS_FILE],
-                    'hbc_results': config[self.identifier][pc.HBC_FILE],
-                    'vaf_results': config[self.identifier][pc.VAF_FILE]
-                }
+        data = self.get_starting_plugin_data(wrapper, self.PLUGIN_VERSION)
+        f_config = configparser.ConfigParser() #Initializing the configparser
```

Comment:
Making a `ConfigParser` here won't work in production, because `full_config.ini` does not exist at configure time -- it is written _after_ all plugins have run their config step, to record the results.

If a plugin needs config parameters, it should either get them from the input to its `configure` method (which is itself a `ConfigParser`), or from files written to the workspace by previously run plugins/helpers.